google/cloud/spanner/transaction.cc

```diff
@@ -97,7 +97,7 @@ Transaction::Transaction(std::string session_id, std::string transaction_id) {
   google::spanner::v1::TransactionSelector selector;
   selector.set_id(std::move(transaction_id));
   impl_ = std::make_shared<internal::TransactionImpl>(
-      internal::MakeDissociatedSessionHolder(std::move(session_id)),
+      internal::MakeDissociatedSessionHolder(std::move(session_id), nullptr),
```

Comment:
At the moment it doesn't look like MakeDissociatedSessionHolder() really needs to be a template.  Might it just take a `std::string session_name` and then be able to provide the `nullptr` to the `Session` ctor itself?